lib/types.nix

```diff
@@ -493,6 +497,31 @@ rec {
       modules = toList modules;
     };
 
+    # The type of a type!
+    type = mkOptionType {
+      name = "type";
+      description = "type";
+      check = isAttrs;
```

Comment:
This should check at least `_type = "option-type";`

Naming isn't quite consistent: `mkOptionType`/`"option-type"`/`types.type`. "Option type" currently wins, but I do think it's a bit verbose. Renaming `mkOptionType` -> `mkType` would also be ok. Just don't deprecate immediately.

Reply:
I think it's fine to be a bit on the verbose side, especially also because `types.type` is kind of weird to see, aligning that with `mkOptionType` should make it a bit more obvious. I'll improve the `check` and change the name to `types.optionType`

Reply:
Also in the module system there's other "types" than just option types, see the other values that `_type` can have. So I think naming this `types.optionType` is very fitting.